include/os/freebsd/spl/sys/atomic.h

```diff
@@ -149,6 +149,23 @@ atomic_cas_64(volatile uint64_t *target, uint64_t cmp, uint64_t newval)
 #endif
 #endif
 
+static inline int
+atomic_dec_not_last_64(volatile uint64_t *target)
+{
+	uint64_t val, oldval;
+
+	val = *target;
+	for (;;) {
+		if (val <= 1) {
+			return (0);
+		}
+		oldval = atomic_cas_64(target, val, val - 1);
+		if (oldval == val)
+			return (1);
+		val = oldval;
+	}
+}
```

Comment:
This behavior doesn't exactly match what the `atomic64_dec_if_positive` call returns on linux.

https://elixir.bootlin.com/linux/latest/source/include/linux/atomic/atomic-instrumented.h#L3129

Reply:
i noticed; i'm going to add more atomic routines (for not bumping above 0) and update the review later